dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -118,8 +118,8 @@ public String readFile(String repositoryId, String fileName, String reference) {
         try {
             repo = github.getRepository(repositoryId);
         } catch (IOException e) {
-            LOG.error(gitUsername + ": IOException on readFile " + e.getMessage());
-            return null;
+            LOG.error(gitUsername + ": IOException on readFile while trying to get the repository " + repositoryId + " " + e.getMessage());
+            throw new CustomWebApplicationException("Could not get repository " + repositoryId + " from GitHub.", HttpStatus.SC_BAD_REQUEST);
```

Comment:
In the past, this might bring down a refresh org or refresh all. But since we're getting rid of those, it *hopefully* doesn't matter.

Reply:
I don't know this code that well, but this seems like a big change. What happens if a workflow imports 3 files, but it only finds two of the files? Before this change, would there be a "partial" version, with some of the files, and now there will be none?

Reply:
Follow up thought -- if we weren't doing a null check when this is being invoked, then this is better than it was.